### PR TITLE
delete-office by Husan

### PR DESCRIPTION
**Pull request by khusanKholmatov**

### CODE REVIEW

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 1]
```diff
@@ -29,4 +29,9 @@
     public List<Office> getAllOffices(){
         return officeService.getAllOffices();
     }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteOfficeById(@PathVariable UUID id){
+        return ResponseEntity.status(HttpStatus.OK).body(officeService.deleteOffice(id));
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:35]: just return 204 http code with no response body

File: src/main/java/com/exadel/coolDesking/office/OfficeDeletedDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class OfficeDeletedDto {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeDeletedDto.java:14]: you don't need it

File: src/main/java/com/exadel/coolDesking/office/OfficeService.java [hunk 3]
```diff
@@ -56,4 +60,14 @@ public OfficeResponse getOfficeById(UUID id) {
     public Optional<Office> getOfficeByName(String officeName) {
         return officeRepository.getOfficeByName(officeName);
     }
+
+    public OfficeDeletedDto deleteOffice(UUID id) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/office/OfficeService.java:64]: void

File: src/main/java/com/exadel/coolDesking/user/UserService.java [hunk 4]
```diff
@@ -27,4 +29,11 @@ public void updateUserStatus(UserState userState, String telegramId) {
                     User.class,
                     telegramId);
     }
+
+    public void deleteUserWorkplaceId(List<User> userList){
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/user/UserService.java:33]: `deleteUserPreferredWorkplaceId` maybe?

File: src/main/resources/db/changelog/changeset/24-drop-office_id-foreign-key-adding-new_one-in-floor-plan.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            foreignKeyName: fk_floor_plan_office
+      changes:
+        - dropForeignKeyConstraint:
+            baseTableName: floor_plan
+            constraintName: fk_floor_plan_office
+        - addForeignKeyConstraint:
+            baseColumnNames: office_id
+            baseTableName: floor_plan
+            deleteCascade: true
+            constraintName: fk_floor_plan_office2
```
Comment (ohalushchak-exadel) [inline src/main/resources/db/changelog/changeset/24-drop-office_id-foreign-key-adding-new_one-in-floor-plan.yaml:17]: why not just `fk_floor_plan_office`?

File: src/main/resources/db/changelog/changeset/24-drop-office_id-foreign-key-adding-new_one-in-floor-plan.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+          foreignKeyConstraintExists:
+            foreignKeyName: fk_floor_plan_office
+      changes:
+        - dropForeignKeyConstraint:
+            baseTableName: floor_plan
+            constraintName: fk_floor_plan_office
+        - addForeignKeyConstraint:
+            baseColumnNames: office_id
+            baseTableName: floor_plan
+            deleteCascade: true
```
Comment (ohalushchak-exadel) [inline src/main/resources/db/changelog/changeset/24-drop-office_id-foreign-key-adding-new_one-in-floor-plan.yaml:16]: `addForeignKeyConstraint` does not support `deleteCascade` attribute.

File: src/main/resources/db/changelog/changeset/25-drop-floor_plan_id-foreign-key-adding-new_one-in-workplace.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+          foreignKeyConstraintExists:
+            foreignKeyName: fk_workplace_floor_plan
+      changes:
+        - dropForeignKeyConstraint:
+            baseTableName: workplace
+            constraintName: fk_workplace_floor_plan
+        - addForeignKeyConstraint:
+            baseColumnNames: floor_plan_id
+            baseTableName: workplace
+            deleteCascade: true
```
Comment (ohalushchak-exadel) [inline src/main/resources/db/changelog/changeset/25-drop-floor_plan_id-foreign-key-adding-new_one-in-workplace.yaml:16]: `addForeignKeyConstraint` does not support `deleteCascade` attribute.

File: src/main/resources/db/changelog/changeset/25-drop-floor_plan_id-foreign-key-adding-new_one-in-workplace.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+            constraintName: fk_workplace_floor_plan2
```
Comment (ohalushchak-exadel) [inline src/main/resources/db/changelog/changeset/25-drop-floor_plan_id-foreign-key-adding-new_one-in-workplace.yaml:17]: why not just `fk_workplace_floor_plan`

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java (hunk 1 above)
Comment (khusanKholmatov) [reply src/main/java/com/exadel/coolDesking/office/OfficeController.java:35]: but, i have to return id, and message according to Google Doc

Comment (ohalushchak-exadel) [reply src/main/java/com/exadel/coolDesking/office/OfficeController.java:35]: that doc is pretty bad, unfortunately

**Review by ohalushchak-exadel — APPROVED** [review]

**Pull request merged**